sm-shop/src/main/java/com/salesmanager/shop/store/controller/order/facade/OrderFacadeImpl.java

```diff
@@ -1259,28 +1259,29 @@ public Order processOrder(com.salesmanager.shop.model.order.v1.PersistableOrder
 			// of process order request. If totals does not match, an error
 			// should be thrown.
 
-			OrderTotalSummary orderTotalSummary = null;
+
 
 			OrderSummary orderSummary = new OrderSummary();
 			orderSummary.setShippingSummary(shippingSummary);
 			List<ShoppingCartItem> itemsSet = new ArrayList<ShoppingCartItem>(cart.getLineItems());
 			orderSummary.setProducts(itemsSet);
 
-			orderTotalSummary = orderService.caculateOrderTotal(orderSummary, customer, store, language);
+			OrderTotalSummary orderTotalSummary = orderService.caculateOrderTotal(orderSummary, customer, store, language);
 
 			if (order.getPayment().getAmount() == null) {
 				throw new ConversionException("Requires Payment.amount");
 			}
 
-			String submitedAmount = order.getPayment().getAmount();
+			BigDecimal submittedAmount = new BigDecimal(order.getPayment().getAmount());
 
 			BigDecimal calculatedAmount = orderTotalSummary.getTotal();
 			String strCalculatedTotal = calculatedAmount.toPlainString();
```

Comment:
`StrCalculatedTotal ` if you will no use it, you could remove it from the code.

Reply:
Done